frontend/CODE_GUIDELINES.md

```diff
@@ -0,0 +1,105 @@
+# Code Guidelines for React
+
+Hat tip to Clean Code by Robert C. Martin (Uncle Bob)
+
+## Goals
+
+- Consistency
+- Predictability
+- Maintainability
+- Readability
+
+## Style
+
+- Take time to craft great variable and function names. It's ok to be verbose. Avoid abbreviations.
+- Replace magic numbers with named constants.
+- Remove dead code aggressively. We can always recover it from version control.
+- Use vertical whitespace to aid readability.
+  - Related lines should be vertically close.
+  - A block of variable declarations should be followed by a blank line.
+  - Use a blank line above return statements.
+  - Don't stack code too tight vertically, especially `if` blocks.
+- Use Prettier/ESLint to avoid bike-shedding.
+- Prefer copy/paste to premature abstraction.
+  - Abstractions tend to reveal themselves quickly on the second or third copy/paste.
+  - It's easier to abstract code than to remove an abstraction.
+- Break switch statements into their own functions with returns rather than `break;`.
```

Comment:
I don't feel too strongly about this, but could be convinced

Reply:
JavaScript STILL doesn't have `do` blocks. This is my pattern for hacking around the lack of `do` blocks. It's divisive. I've had some coworkers who've fallen in love with it, and other who think I'm ridiculous for using `switch` so much.